### PR TITLE
cdap solution template for v2 marketplace

### PR DESCRIPTION
**Pull request by dereklwood**
solution template for deploying CDAP on the Azure HDInsight v2 marketplace platform.  Most of this is boilerplate, with the CDAP-specific bits customized.  This should be merged with https://github.com/caskdata/cdap/pull/5985
- [x] set to use cdap `release/3.4` branch
- [ ] httpsEndpoints for UI doesn't actually work yet... working with them on it.

### CODE REVIEW

File: cdap-distributions/src/hdinsight/cdap-deploy.json [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                    "name": "edgenode",
+                    "targetInstanceCount": 1,
+                    "hardwareProfile": {
+                        "vmSize": "[parameters('edgeNodeSize')]"
+                    }
+                }]
+            },
+            "installScriptActions": [{
+                "name": "[concat('cdap-pageblob-configure-v0','-' ,uniquestring(variables('applicationName')))]",
+                "uri": "https://raw.githubusercontent.com/caskdata/cdap/release/3.4/cdap-distributions/src/hdinsight/pageblob-configure.sh",
```
Comment (wolf31o2) [inline cdap-distributions/src/hdinsight/cdap-deploy.json:53]: Should we publish this somewhere, versus using GitHub directly?

**Comment by wolf31o2** [comment]
One question, but otherwise 👍 LGTM

File: cdap-distributions/src/hdinsight/cdap-deploy.json (hunk 1 above)
Comment (dereklwood) [inline cdap-distributions/src/hdinsight/cdap-deploy.json:53]: Yeah, we can still release the scripts as artifacts somewhere, as we had talked about doing for v1.  Until we know the location though, this makes it easier to test.  So yes these will need to change, but we can discuss internally where they should go.

**Comment by dereklwood** [comment]
travis unrelated... merging

**Pull request merged**